torch/csrc/autograd/engine.cpp

```diff
@@ -1336,6 +1335,7 @@ void Engine::init_local_ready_queue(std::shared_ptr<ReadyQueue> ready_queue) {
 auto Engine::ready_queue(
     std::shared_ptr<ReadyQueue> cpu_ready_queue,
     at::Device device) -> std::shared_ptr<ReadyQueue> {
+  initialize_device_threads_pool();
```

Comment:
Not really no.
Because this will always initialize all the threads. So that doesn't change much.

The idea was to initialize for each device individually indeed.
And yes it will require quite a bit of refactor (that's why I created an issue instead of sending a PR).

Reply:
Got it. Thanks.